Documentation/test-plans/men-7729-zephyr-onboarding.md

```diff
@@ -0,0 +1,178 @@
+# MEN-7729 Zephyr onboarding
+
+## Summary
+
+This test plan covers the implementation of the Zephyr onboarding functionality.
+
+This project introduces a new "Micro" and "Standard" device tiers. Micro for microcontrollers and a "Standard" tier for anything else.
+The Micro tier aims to optimize for devices with lower software complexity and smaller artifact sizes.
+The Standard tier will be the default tier and used for all devices without any other tier specified.
+
+This project also enables new users to get started with the Zephyr platform through a guided setup process.
+
+## Related
+
+- [MEN-7729 Zephyr onboarding (Epic)](https://northerntech.atlassian.net/browse/MEN-7729)
+- [User experience for Zephyr (Docs)](https://docs.google.com/document/d/1-3BtSTl5kr4lT9Rcah-G4cRQhgusr0VmRIYmWVE3Ygg)
+- [Pricing model for MCUs (Docs)](https://docs.google.com/document/d/1R8gUejuQecvYKuJlq0LEVzHFdMC3MSbOTHXZXaWD3Bo)
+- [UX/UI (Figma)](https://www.figma.com/design/ZTGKMcKtHrZ0V7pfdws7KM/Zephyr-onboarding-UX-concept?node-id=3761-2931&p=f&t=4EWRNIcCrseZ054D-0)
+- API documentation TBA
+- Onboarding (Docs) TBA
+
+## Features breakdown
+
+### Device tiers
+
+- Two device tiers: "Micro" and "Standard"
+    - Orthogonal to the current plan types (Trial, Basic, Professional, Enterprise)
+    - All tenants will be able to have device limits for both tiers
+- Commercial plans (Basic, Professional, Enterprise) with Micro tier devices will, by default, have a maximum 5 MB Artifact size for deployment
+- Device tier will be configured on the device side and included as part of the authentication set
+- During plan purchase, users can select the number of devices to purchase for each tier
+- Device tier limit will be synchronized with HubSpot
+
+### Onboarding process
+
+- Users can choose an MCU with Zephyr among the devices options
+- Guided process for initial device setup
+- For MCUs, the reference device will be Espressif ESP32-S3
+- New "Microcontroller Get started" tutorial
+
+### UX
+
+- A new device type (Zephyr/MCU) is available for onboarding
+- The UI will provide a configuration snippet for building MCU images
+    - the token includes the device tier and tenant token
+- A warning will be displayed in the UI if an artifact is too large for the "Micro" tier during deployment
```

Comment:
Note that this is opportunistic, as we can't always know in advance which devices will be part of a deployment in case of dynamic deployments.. So maybe append "when technically possible"